### ISSUE TITLE
New-M365DSCDeltaReport completes but generates ArgumentTransformationMetadataException error

### ISSUE THREAD

**Issue by Kikaider** [microsoft/Microsoft365DSC#2260]
#### Details of the scenario you tried and the problem that is occurring
When executing New-M365DSCDeltaReport locally and via Azure Runbook, the cmdlet generates an [ArgumentTransformationMetadataException] exception, but does generate the html report. I tried the cmdlet with M365DSC extracts of several different M365DSC components with the same error every time.

`New-M365DSCDeltaReport -Source "runbook_dscconfig.ps1" -Destination "runbook_export.ps1" -OutputPath $null | Out-File -FilePath "runbook_2022-Sep-07-0652AM.html"`

#### Verbose logs showing the problem
```
Cannot convert the "Loading file 'E:\M365DSC\M365x758710\M365DSC\tmp\runbook_export.ps1'" value of type "System.String" to type "System.Collections.Hashtable".
At C:\Program Files\WindowsPowerShell\Modules\Microsoft365DSC\1.22.831.1\modules\M365DSCReport.psm1:598 char:14
+     foreach ($destinationResource in $DestinationObject)
+              ~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : MetadataError: (:) [], ArgumentTransformationMetadataException
    + FullyQualifiedErrorId : RuntimeException
```
#### Suggested solution to the issue
Workaround is to capture the [ArgumentTransformationMetadataException] exception and ignore it

```
try
{
    New-M365DSCDeltaReport -Source $sourceFilename -Destination $exportFilename -OutputPath $null | Out-File -FilePath $deltaFilename -Encoding ascii
}
catch [System.Management.Automation.ArgumentTransformationMetadataException]
{
    Write-Host "Gotcha!"
}
```

#### The DSC configuration that is used to reproduce the issue (as detailed as possible)
See attached files (change extension back to .ps1) for New-M365DSCDeltaReport cmdlet.

#### The operating system the target node is running
N/A

#### Version of the DSC module that was used ('dev' if using current dev branch)
1.22.831.1
[runbook_export .txt](https://github.com/microsoft/Microsoft365DSC/files/9522025/runbook_export.txt)
[runbook_dscconfig.txt](https://github.com/microsoft/Microsoft365DSC/files/9522026/runbook_dscconfig.txt)

**Comment by andikrueger** [comment]
This is the passage causing the error message in the verbose log:

https://github.com/microsoft/Microsoft365DSC/blob/320ab9d213cfb5f0515783e29b1a97a861be2aa1/Modules/Microsoft365DSC/Modules/M365DSCReport.psm1#L620-L660

Could you run a test with two resources within your configuration? This looks like there is the need to change the code a bit to have proper arrays and not only to cast the objects.

**Comment by Kikaider** [comment]
Same error with 1 component, 2 components, and 53 components.

`$Components = "SPOSharingSettings"`

`$Components = @("SPOSharingSettings","SPOAccessControlSettings")`

`$Components = @("AADApplication", "AADAuthorizationPolicy", "AADConditionalAccessPolicy", "AADGroup", "AADGroupLifecyclePolicy", "AADGroupsNamingPolicy", "AADGroupsSettings", "AADNamedLocationPolicy", "AADRoleDefinition", "AADSecurityDefaults", "AADTenantDetails", "AADTokenLifetimePolicy", "IntuneAntivirusPolicyWindows10SettingCatalog", "IntuneAppConfigurationPolicy", "IntuneApplicationControlPolicyWindows10", "IntuneAppProtectionPolicyAndroid", "IntuneAppProtectionPolicyiOS", "IntuneASRRulesPolicyWindows10", "IntuneAttackSurfaceReductionRulesPolicyWindows10ConfigManager", "IntuneDeviceAndAppManagementAssignmentFilter", "IntuneDeviceCategory", "IntuneDeviceCompliancePolicyAndroid", "IntuneDeviceCompliancePolicyAndroidDeviceOwner", "IntuneDeviceCompliancePolicyAndroidWorkProfile", "IntuneDeviceCompliancePolicyiOs", "IntuneDeviceCompliancePolicyMacOS", "IntuneDeviceCompliancePolicyWindows10", "IntuneDeviceConfigurationPolicyAndroidDeviceOwner", "IntuneDeviceConfigurationPolicyAndroidWorkProfile", "IntuneDeviceConfigurationPolicyiOS", "IntuneDeviceConfigurationPolicyWindows10", "IntuneDeviceEnrollmentLimitRestriction", "IntuneDeviceEnrollmentPlatformRestriction", "IntuneExploitProtectionPolicyWindows10SettingCatalog", "IntuneSettingCatalogASRRulesPolicyWindows10", "ODSettings", "SPOAccessControlSettings", "SPOApp", "SPOBrowserIdleSignout", "SPOHomeSite", "SPOHubSite", "SPOOrgAssetsLibrary", "SPOSearchManagedProperty", "SPOSearchResultSource", "SPOSharingSettings", "SPOSiteDesign", "SPOSiteDesignRights", "SPOSiteScript", "SPOStorageEntity", "SPOTenantCdnEnabled", "SPOTenantCdnPolicy", "SPOTenantSettings", "SPOTheme")`

**Comment by Kikaider** [comment]
Seems to be fixed in v1.22.914.1
